### DISCUSSION TITLE
I'm confused about the documentation for WASDK.

### DISCUSSION THREAD

**Discussion by themerror** [microsoft/WindowsAppSDK#2680] [General] [answered]
Hello.

I'm trying to get my battery information via WinUI 3. Then, I find this [Devices and sensors](https://docs.microsoft.com/en-us/windows/apps/develop/devices-and-sensors). It says 'The [Windows App SDK](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/) currently does not provide APIs related to devices and sensors scenarios.' But, I'm navigated to  [Get battery information](https://docs.microsoft.com/en-us/windows/uwp/devices-sensors/get-battery-info) following links in Windows OS features.
So what is the relationship between Windows SDK and Windows App SDK? Will Microsoft make some Windows SDK features a part of WASDK? And what is a winmd file? Does WinUI 3 rely on this?

It confused me a lot when I use WinRT APIs.

**Comment by dhoehna** [comment, marked as answer]
Hi themerror,

Thank you for reaching out.  Hopefully I can answer some of your questions.

> So what is the relationship between Windows SDK and Windows App SDK?
The Windows SDK is for accessing the bare windows APIs like Registry API's and IShellLinkItems and other headers that are needed to make a program that interacts with windows specific features.

The Windows App SDK, how I see it, is a layer above the windows APIs.  For example, Environment Manager.  Environment manager is an abstraction layer above the registry APIs, that lets developers quickly edit environment variables without having to bother with directly calling, and subsequently, learning, the registry APIs.  [Here is a more formal explanation of what Windows App SDK is](https://github.com/microsoft/WindowsAppSDK/blob/main/docs/README.md)

What Windows App SDK is not.  Windows App SDK is not a replacement for the Windows SDK.

> Will Microsoft make some Windows SDK features a part of WASDK? 
Maybe.  I haven't heard anything.  Is there a reason why you want something ported from the SDK to Windows App SDK?

> What is a winmd file?
A Winmd file is a metadata file that WinRT uses to expose information about itself to other code that wants to consume it.  For example, a C++ class has a header file and the header file documents what is publicly available in the class.  A winmd file is the same thing, just for WinRT.

> Does WinUI 3 rely on this?
Because WinUI3 is using CppWinRT, yes.  WinUI uses winmd files.

> It confused me a lot when I use WinRT APIs.
I totally, 100%, understand.  I was also confused about WinRT when I first started developing for, and using it.  Is there something specific that is confusing?

Hopefully that answered some of your questions.  Please keep them coming if you have more.

**Reply by themerror** [reply]
Thanks for your answer.
So when I use WASDK, actually I'm usingboth Win 32 APIs and WinRT APIs, which are provided by WASDK and/or Windows SDK(UWP), right?

**Reply by contextfree** [reply]
yep, that's right.

**Comment by castorix** [comment]
[Windows SDK](https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/) is the old Platform SDK, the Libraries and Headers for Win32 apps
WinMD files are described in MSDN : [Windows Metadata (WinMD) files](https://docs.microsoft.com/en-us/uwp/winrt-cref/winmd-files)
In C#, you can use [CsWinRT](https://github.com/microsoft/CsWinRT) to use them (in the .csproj, mainly with _CsWinRTIncludes_)
